pkg/controller/job/job_controller.go

```diff
@@ -900,6 +900,7 @@ func activePodsForRemoval(job *batch.Job, pods []*v1.Pod, rmAtLeast int) []*v1.P
 		rm = make([]*v1.Pod, 0, rmAtLeast)
 		left = make([]*v1.Pod, 0, len(pods)-rmAtLeast)
 		rm, left = appendDuplicatedIndexPodsForRemoval(rm, left, pods)
+		rm = appendPodsWithInvalidIndexForRemoval(rm, left, int(*job.Spec.Completions))
```

Comment:
we need to know which pods were not removed. Perhaps you can incorporate the logic in `appendDuplicatedIndexPodsForRemoval`